src/rest/json.cpp

```diff
@@ -0,0 +1,539 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rest/json.hpp"
+
+#include "common/code_utils.hpp"
+
+extern "C" {
+#include "cJSON.h"
+}
+
+namespace otbr {
+namespace rest {
+namespace JSON {
+
+cJSON *Bytes2HexJson(const uint8_t *aBytes, uint8_t aLength)
+{
+    char   hex[2 * aLength + 1];
+    cJSON *ret = nullptr;
+
+    otbr::Utils::Bytes2Hex(aBytes, aLength, hex);
+    hex[2 * aLength] = '\0';
+    ret              = cJSON_CreateString(hex);
+
+    return ret;
+}
+
+std::string String2JsonString(std::string aString)
+{
+    std::string ret;
+    cJSON *     json    = nullptr;
+    char *      jsonOut = nullptr;
+
+    VerifyOrExit(aString.size() > 0);
+
+    json    = cJSON_CreateString(aString.c_str());
+    jsonOut = cJSON_Print(json);
+    ret     = jsonOut;
+    delete (jsonOut);
```

Comment:
Should use `cJSON_free` which is bind to `free` by default.

> It will allocate a string and print a JSON representation of the tree into it. Once it returns, you are fully responsible for deallocating it after use with your allocator. (usually free, depends on what has been set with cJSON_InitHooks).